aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -217,7 +217,7 @@ struct exponential_distribution {
   inline T operator()(RNG* generator) {
     // Follows numpy exponential for the case when lambda is zero.
     if (lambda == static_cast<T>(0.0)) {
-      return static_cast<T>(0.0);
+      return std::numeric_limits<T>::infinity();
```

Comment:
in the future we could also try to run the calculation and see if it works out.

Reply:
(assuming the below doesn't do any undefined behavior, which I didn't check).

Reply:
Correct. If this isn't urgent for 1.5. I'll revisit just removing the check.